From 72a364125528fa533cfddb751018ab30bb12dafb Mon Sep 17 00:00:00 2001
From: Holger Freyther <zecke@selfish.org>
Date: Wed, 26 Sep 2012 17:22:30 +0200
Subject: [PATCH 1/4] qlibraryinfo: allow to set qt.conf from the outside using
 the environment

Allow to set a qt.conf from the outside using the environment. This allows
to inject new prefixes and other paths into qmake. This is needed when using
the same qmake binary to build qt/x11 and qt/embedded

Upstream-Status: Pending

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/corelib/global/qlibraryinfo.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/corelib/global/qlibraryinfo.cpp b/src/corelib/global/qlibraryinfo.cpp
index 747fd87..c9f8170 100644
--- a/src/corelib/global/qlibraryinfo.cpp
+++ b/src/corelib/global/qlibraryinfo.cpp
@@ -140,6 +140,10 @@ QSettings *QLibraryInfoPrivate::findConfiguration()
 {
     QString qtconfig = QStringLiteral(":/qt/etc/qt.conf");
 #ifdef QT_BOOTSTRAPPED
+    if (!QFile::exists(qtconfig)) {
+        QByteArray config = getenv("QT_CONF_PATH");
+        qtconfig = QFile::decodeName(config);
+    }
     if(!QFile::exists(qtconfig))
         qtconfig = qt_libraryInfoFile();
 #else
@@ -164,6 +168,10 @@ QSettings *QLibraryInfoPrivate::findConfiguration()
 	    }
     }
 #endif
+    if (!QFile::exists(qtconfig)) {
+        QByteArray config = getenv("QT_CONF_PATH");
+        qtconfig = QFile::decodeName(config);
+    }
     if (QFile::exists(qtconfig))
         return new QSettings(qtconfig, QSettings::IniFormat);
     return 0;     //no luck
-- 
1.8.1.5

